### DISCUSSION TITLE
How can I stop plan?

### DISCUSSION THREAD

**Discussion by AlekseySVTN** [rancher/system-upgrade-controller#208] [General]
Hello to all,

Sorry for the maybe dumb question.
Right now I testing system upgrade controller.
And I would like to know, how I can stop execution of my plan after failure.
So, for example
-- I have a plan
-- plan have a job
-- job have a failure state
-- after ttlSecondsAfterFinished (by default - 900) this job will delete, and plan will create a new job with the same options
In my case I do not want that plan will recreate failure job
How can I do this?
May be I can set ttlSecondsAfterFinished to max size using ENVIROVMENT or may be another way is exist?

Thank you for attention

**Comment by dweomer** [comment]
> May be I can set ttlSecondsAfterFinished to max size using ENVIROVMENT or may be another way is exist?

Yes, if you bump that value in the envvars for the controller your failed jobs will stick around longer.

Another thing that you can try is to stop the SUC pod to prevent job recreation until you're ready.
